### ISSUE TITLE
Pagination showing blank pages for hidden products

### ISSUE THREAD

**Issue by barbedhook** [woocommerce/woocommerce#13976]
### EXPLANATION OF THE ISSUE

Woocommerce 3.0 with Storefront theme. Pagination is showing pages for every products even if it is "hidden when out of stock". Product isn't showing, just lots of blank pages when viewing by category. 

### STEPS TO REPRODUCE THE ISSUE

Updated to 3.0

### SYSTEM STATUS

<details>
`
### WordPress Environment ###

Home URL: http://www.pintailwaterfowl.com
Site URL: http://www.pintailwaterfowl.com
WC Version: 3.0.0
Log Directory Writable: ✔
WP Version: 4.7.3
WP Multisite: –
WP Memory Limit: 512 MB
WP Debug Mode: –
WP Cron: ✔
Language: en_US

### Server Environment ###

Server Info: Apache/2.2.32 (Unix) mod_ssl/2.2.32 OpenSSL/1.0.1e-fips mod_bwlimited/1.4
PHP Version: 5.6.30
PHP Post Max Size: 500 MB
PHP Time Limit: 30
PHP Max Input Vars: 1000
cURL Version: 7.38.0
OpenSSL/1.0.1e

SUHOSIN Installed: –
MySQL Version: ❌ 5.5.54 - We recommend a minimum MySQL version of 5.6. See: WordPress requirements
Max Upload Size: 500 MB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ✔
DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

WC Database Version: 3.0.0
WC Database Prefix: wp_
woocommerce_sessions: ✔
woocommerce_api_keys: ✔
woocommerce_attribute_taxonomies: ✔
woocommerce_downloadable_product_permissions: ✔
woocommerce_order_items: ✔
woocommerce_order_itemmeta: ✔
woocommerce_tax_rates: ✔
woocommerce_tax_rate_locations: ✔
woocommerce_shipping_zones: ✔
woocommerce_shipping_zone_locations: ✔
woocommerce_shipping_zone_methods: ✔
woocommerce_payment_tokens: ✔
woocommerce_payment_tokenmeta: ✔

### Security ###

Secure connection (HTTPS): ❌Your store is not using HTTPS. Learn more about HTTPS and SSL Certificates.
Hide errors from visitors: ✔

### Active Plugins (32) ###

WooCommerce ShippingEasy Integration: by WooThemes – 3.4.1
Social Media Widget by Acurax: by Acurax – 3.2.3
Advanced Database Cleaner: by Younes JFR. – 2.0.0
Akismet Anti-Spam: by Automattic – 3.3
All In One SEO Pack: by Michael Torbert – 2.3.12.1
iThemes Security: by iThemes – 6.2.0
Better WordPress Google XML Sitemaps: by Khang Minh – 1.4.1
Cleverness To-Do List: by C.M. Kendrick – 3.4.2
Custom Post Type UI: by WebDevStudios – 1.5.2
Email Users: by Mike Walsh & MarvinLabs – 4.8.4
Google Analyticator: by SumoMe – 6.5.2
Image Rotation Repair: by Shane & Peter
Inc. – 1.1

Image Watermark: by dFactory – 1.6.1
jQuery Colorbox: by Arne Franken – 4.6.2
jQuery Updater: by Ramoonus – 3.2.0
More Fields: by Henrik Melin
Kal Ström – 2.1

P3 (Plugin Performance Profiler): by GoDaddy.com – 1.5.3.9
VarkTech Pricing Deals for WooCommerce: by Vark – 1.1.6.7
Fast Secure Contact Form: by Mike Challis
Ken Carlson – 4.0.50

SSL Insecure Content Fixer: by WebAware – 2.2.3
Use Google Libraries: by Jason Penney – 1.6.2.1
WooCommerce Bulk Discount: by Rene Puchinger – 2.4.4
WooCommerce Cart Image - Widget and Shortcode - Wordpress Sidebar and Posts Plugin: by WooCommerce and Wordpress Plugins – 1.0
WooCommerce Cost of Goods: by WooThemes / SkyVerge – 1.8.0 – 2.3.0 is available
WooCommerce Custom Status: by Garman Technical Services – 1.0
WooCommerce Print Invoice & Delivery Note: by Triggvy Gunderson – 4.1.5
WooCommerce Product Add-ons: by WooThemes – 2.7.9 – 2.8.1 is available
WooCommerce Store Credit: by WooThemes – 2.1.5
WooCommerce: by Automattic – 3.0.0
WordPress Gzip Compression: by James Socol – 1.0
WP Super Cache: by Automattic – 1.4.9
YITH WooCommerce Affiliates: by YITHEMES – 1.0.8

### Settings ###

API Enabled: ✔
Force SSL: –
Currency: USD ($)
Currency Position: left
Thousand Separator: ,
Decimal Separator: .
Number of Decimals: 2
Taxonomies: Product Types: external (external)
grouped (grouped)
simple (simple)
variable (variable)

### WC Pages ###

Shop base: #363 - /shop/
Cart: #364 - /cart/
Checkout: #365 - /checkout/
My account: #367 - /my-account/

### Theme ###

Name: Pintail Waterfowl Theme
Version: 1.0.0
Author URL: http://www.barbedhook.com
Child Theme: ✔
Parent Theme Name: Storefront
Parent Theme Version: 2.1.8
Parent Theme Author URL: https://woocommerce.com/
WooCommerce Support: ✔

### Templates ###

Overrides: –
`
</details>

**Comment by mikejolley** [comment]
Let's not ask in multiple places https://wordpress.org/support/topic/3-0-issues-pagination-terms-conditions/

**Comment by yratof** [comment]
I've noticed that the discussion for this issue was overtaken by another issue on the woocommerce support. Maybe this is a better place for localised problem solving?

I've found that updating a multisite to v.3 has resulted in a similar issue.

Products that are out of stock seem to be showing up within the pagination. The database was updated, I've used the recalculate terms button within the setting — Have you encountered this problem yet?
